packages/browser-agent-core/src/features/spa/aggregate/index.js

```diff
@@ -422,6 +424,20 @@ export class Aggregate extends FeatureBase {
     register('newURL', function (url, hashChangedDuringCb) {
       if (state.currentNode) {
         state.currentNode[INTERACTION].setNewURL(url)
+      } else if (state.prevInteraction && !state.prevInteraction.ignored) {
+        /*
+         * The previous interaction was discarded before the route was changed. This can happen in SPA
+         * frameworks when using lazy loading. We have also seen this in version 11+ of Nextjs where
+         * some route changes re-use cached resolved promises.
+         */
+        const interaction = state.prevInteraction
+        interaction.setNewURL(url)
+        interaction.root.end = null
+
+        setCurrentNode(interaction.root)
```

Comment:
This may be addressed by your update to ln699-704, but if `state.prevInteraction` is retained, would there be any workflow in which a URL change is incorrectly assigned to that prevInteraction?

Reply:
Should not be. Lines 699-704 should take care of it. Thats why those lines were added.